eth/downloader/modes.go

```diff
@@ -23,4 +23,5 @@ const (
 	FullSync  SyncMode = iota // Synchronise the entire blockchain history from full blocks
 	FastSync                  // Quickly download the headers, full sync only at the chain head
 	LightSync                 // Download only the headers and terminate afterwards
+	NoSync                    // Dummy downloader, only one of the eth/les downloaders is actually working
```

Comment:
Please remove **all** notions of this. It's very very bad design. If you don't want the downloader to run, don't start it. This is just crippling the entire code base.